src/adapter/src/coord/timestamp_selection.rs

```diff
@@ -111,13 +111,29 @@ impl<S: Append + 'static> Coordinator<S> {
             ));
         };
 
-        Ok(TimestampDetermination {
+        let det = TimestampDetermination {
             timestamp,
             since,
             upper,
             largest_not_in_advance_of_upper,
             oracle_read_ts,
-        })
+        };
+        self.metrics
+            .determine_timestamp
+            .with_label_values(&[
+                match det.respond_immediately() {
+                    true => "true",
+                    false => "false",
+                },
+                isolation_level.as_str(),
+                // TODO: Should these be bucketed by the full compute_instance?
```

Comment:
As in have a separate bucket for each compute instance? Are these metrics aggregated across all environments? If so then that could get extremely noisy. If not I think that makes sense. There's no inherit performance difference between system and user clusters so it's a bit of an arbitrary distinction to make for metrics. It would also make it easier to identify misbehaving clusters.

Reply:
I think compute id makes more sense. If we are looking at metrics globally we’d probably roll up to the environment id anyway

Reply:
Changed to the full id.